protocols/kad/src/behaviour.rs

```diff
@@ -593,6 +597,9 @@ where
         peer: &PeerId,
         address: &Multiaddr,
     ) -> Option<kbucket::EntryView<kbucket::Key<PeerId>, Addresses>> {
+        let Ok(address) = &address.to_owned().with_p2p(*peer) else {
+            return None;
```

Comment:
You should be able to use `.ok()?` here.

Reply:
Indeed. I'm fixing that.